lib/user.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'database_connection'
+require 'bcrypt'
+
+class User
+  include BCrypt
+  attr_reader :id, :name, :username, :email, :password
+
+  def initialize(id:, name:, username:, email:, password:)
+    @id = id
+    @name = name
+    @username = username
+    @email = email
+    @password = password
+  end
+ 
+  def self.create(name:, username:, email:, password:)
+    hidden_password = BCrypt::Password.create(password)
```

Comment:
Could wrap the password class in your own class - that way if you decide to use a different password hash function in the future no need to change everything in the code